File: flutter/digit-ui-components/storybook/lib/widgets/atoms/checkbox_stories.dart
import 'package:flutter/material.dart';
import 'package:digit_ui_components/widgets/atoms/digit_checkbox.dart';
import 'package:storybook_toolkit/storybook_toolkit.dart';
import '../../iframe/iframe_widget.dart';

List<Story> checkboxStories() {
  return [
    Story(
      name: 'Atom/Checkbox/Documentation',
      builder: (context) {
        return IframeWidget(
          url: 'https://core.digit.org/guides/developer-guide/ui-developer-guide/digit-ui/ui-components-standardisation/digit-ui-components-v0.2.0',
        );
      },
    ),
    Story(
      name: 'Atom/Checkbox/Unchecked',
      builder: (context) {
        return Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            Center(
              child: DigitCheckbox(
                value: false,
                label: context.knobs.text(
                  label: 'Label',
                  initial: 'checkbox label',
                ),
                onChanged: (value) {},
                readOnly: context.knobs.boolean(
                  label: 'Read Only',
                  initial: false,
                ),
                isDisabled: context.knobs.boolean(
                  label: 'Disabled',
                  initial: false,
                ),
                alignRight: context.knobs.boolean(label: 'Align Right', initial: false),
                isRequired: context.knobs.boolean(label: 'Required', initial: false),
              ),
            ),
          ],
        );
      },
    ),
    Story(
      name: 'Atom/Checkbox/Checked',
      builder: (context) {
        return Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            Center(
              child: DigitCheckbox(
                label: context.knobs.text(
                  label: 'Label',
                  initial: 'checkbox label',
                ),
                value: true,
                onChanged: (value) {},
                readOnly: context.knobs.boolean(
                  label: 'Read Only',
                  initial: false,
                ),
                isDisabled: context.knobs.boolean(
                  label: 'Disabled',
                  initial: false,
                ),
                alignRight: context.knobs.boolean(label: 'Align Right', initial: false),
              ),
            ),
          ],
        );
      },
    ),
    Story(
      name: 'Atom/Checkbox/Intermediate',
      builder: (context) {
        return Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            Center(
              child: DigitCheckbox(
                label: context.knobs.text(
                  label: 'Label',
                  initial: 'checkbox label',
                ),
                onChanged: (value) {},
                readOnly: context.knobs.boolean(
                  label: 'Read Only',
                  initial: false,
                ),
                isDisabled: context.knobs.boolean(
                  label: 'Disabled',
                  initial: false,
                ),
                alignRight: context.knobs.boolean(label: 'Align Right', initial: false),
              ),
            ),
          ],
        );
      },
    ),
  ];
}

File: react/ui-components/src/atoms/stories/Checkbox.stories.js
import React from "react";
import CheckBox from "../CheckBox";
import Iframe from "../Iframe";

export default {
  title: "Atoms/CheckBox",
  component: CheckBox,
  argTypes: {
    onChange: { action: "onChange", table: { disable: true } },
    value: { control: "text", table: { disable: true } },
    checked: { control: "boolean", table: { disable: true } },
    isIntermediate: { control: "boolean", table: { disable: true } },
    styles: { control: "object", table: { disable: true } },
    style: { control: "object", table: { disable: true } },
    ref: { table: { disable: true } },
    userType: { table: { disable: true } },
    hideLabel: { table: { disable: true } },
    disabled: {
      control: "select",
      options: ["Default","Disabled"],
      name:"state",
      mapping: {
        Default: false, 
        Disabled: true, 
      },
    },
    isLabelFirst: { control: "boolean" },
    label: { control: "text"},
  },
};

const commonStyles = {
  position: "absolute",
  top: "50%",
  left: "50%",
  color: "#363636",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  transform: "translate(-50%, -50%)",
};

const Template = (args) => {
  const [isChecked, setIsChecked] = React.useState(args?.checked);

  const handleCheckboxChange = (event) => {
    setIsChecked(event.target.checked);
  };

  return (
    <div style={commonStyles}>
      <CheckBox
        {...args}
        checked={isChecked}
        onChange={handleCheckboxChange}
      ></CheckBox>
    </div>
  );
};

const commonArgs = {
  onChange: () => {},
  value: "",
  checked: true,
  isLabelFirst: false,
  label: "Label",
  isIntermediate: false,
  styles: {},
  style: {},
  disabled: "Default",
};

export const Documentation = () => (
  <Iframe
    //Todo:Update the url
    src="https://core.digit.org/guides/developer-guide/ui-developer-guide/digit-ui/ui-components-standardisation/digit-ui-core-react-components/input-field"
    title="Checkbox Documentation"
  />
);

Documentation.storyName = "Docs";

export const UnChecked = Template.bind({});
UnChecked.args = {
  ...commonArgs,
  checked:false
};

export const Intermediate = Template.bind({});
Intermediate.args = {
  ...commonArgs,
  checked:false,
  isIntermediate:true
};

export const Checked = Template.bind({});
Checked.args = {
  ...commonArgs,
  checked:true
};